Introducing okta cli

dogeared opened the pull request:
This blog post has the following features:

- [x] A title that's approved by Dev Marketing
- [x] A URL that's approved by Dev Marketing
- [X] Rendered locally and confirmed that no Markdown typos exist

bdemers commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
````diff
@@ -0,0 +1,319 @@
+
+I want to make this a little more challenging for myself, so I am going to use a .NET sample application, since I am a Java developer. Here's the output of the above command:
+
+```bash
````
```suggestion
```txt
```

bdemers commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+
+First I see a one-time requirement in order to provision the Okta org.
+
+I put in some minimal information: name, email and company. In about 20 seconds, a shiny new Okta org is allocated for me and the Okta CLI shows me my Org URL (https://dev-159750.okta.com).
```
```suggestion
I put in some minimal information: name, email and company. In about 20 seconds, a shiny new Okta org is allocated for me and the Okta CLI shows me my Org URL (`https://dev-159750.okta.com`).
```

bdemers commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+
+Once my email address is verified, the Okta CLI displays a unique link to set my password.
+
+> **NOTE:** The order-of-operations here is not important. You can set your password before you choose a sample or afterwards. But, you should set your password in a timely fashion as that unique URL expires after 24 hours.
```
Good call out!

bdemers commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+
+> **NOTE:** The order-of-operations here is not important. You can set your password before you choose a sample or afterwards. But, you should set your password in a timely fashion as that unique URL expires after 24 hours.
+
+Next, I see a list of available samples. As of this writing, there are four samples, but we are expanding that list. And, the Okta CLI automatically makes any new samples available in the list. You can create as many sample as you like with your Okta Org. I chose option 3: `ASP.net Core 4 MVC`
```
As man examples as you like, as long at it's <=5...

bdemers commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+
+Next, I see a list of available samples. As of this writing, there are four samples, but we are expanding that list. And, the Okta CLI automatically makes any new samples available in the list. You can create as many sample as you like with your Okta Org. I chose option 3: `ASP.net Core 4 MVC`
+
+Finally, I see information about the Application that was provisioned in my Okta Org along with instructions to run the sample app.
```
```suggestion
Finally, I see information about the OIDC Application that was provisioned in my Okta Org along with instructions to run the sample app.
```

bdemers commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
````diff
@@ -0,0 +1,319 @@
+```
+
+> Did you know you could run dotnet on macOS? What a world! Simply run: `brew cask install dotnet-sdk`
+
````
Maybe mention a private/incognito window, to make it more obvious the user has logged in.

bdemers commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+
+## Looking Under the Hood of the Okta CLI
+
+I don't know much at all about .NET programming. I couldn't tell you the difference between .NET Core and ASP.NET or the differences between 3 and 4 (I think there's a 5 coming?) But, in less than 5 minutes, I had a running app that I could authenticate to using Okta.
```
This is a great statement!

bdemers commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+
+The important bits to set up a project for use with the Okta CLI are:
+
+1. Add the `OKTA_CLI_` placeholder values to the configuration files that make sense for your language and framework
```
```suggestion
1. Add the `OKTA_CLI_*` placeholder values to the configuration files that make sense for your language and framework
```

bdemers reviewed (approved):
A few minor nits

bdemers commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+    "devops"
+]
+description: "Use the Okt CLI to get started in under 5 minutes!"
+tags: [security, oauth, oauth2, oidc, openid-connect]
```
```suggestion
tags: [security, oauth, oauth2, oidc, openid-connect, cli]
```

mraible commented:
@dogeared Do you still plan on publishing this at some point? If so, we should get it scheduled!

mraible commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+2. Add the `.okta/sample-config.yaml` file with the right application type for your language and framework
+3. Run `okta start` to provision and Okta app and update all the placeholders in your config file(s).
+
+> The easiest way to see what configuration you need for your language and framework is to visit the official samples found in the [okta-samples](https://github.com/okta-samples) GitHub org.
```
"official" samples or "community" samples? 🙂

mraible commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+|------------------------|----------------------------------------------------------------------------------------------------------|
+| CLI_OKTA_ORG_URL       | The base URL to your Okta Org                                                                            |
+| CLI_OKTA_ISSUER        | The URL to your Okta OAuth 2.0 [Authorization Server](https://developer.okta.com/docs/concepts/auth-servers/) |
+| CLI_OKTA_CLIENT_ID     | The client id set by Okta for [web](https://developer.okta.com/docs/guides/implement-auth-code/overview/), [native](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/), [browser](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/) and [service](https://developer.okta.com/docs/guides/implement-client-creds/overview/) type app ...
```
```suggestion
| CLI_OKTA_CLIENT_ID     | The client ID set by Okta for [web](https://developer.okta.com/docs/guides/implement-auth-code/overview/), [native](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/), [browser](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/) and [service](https://developer.okta.com/docs/guides/implement-client-creds/overview/) type apps                                 |
```

mraible commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+| CLI_OKTA_ISSUER        | The URL to your Okta OAuth 2.0 [Authorization Server](https://developer.okta.com/docs/concepts/auth-servers/) |
+| CLI_OKTA_CLIENT_ID     | The client id set by Okta for [web](https://developer.okta.com/docs/guides/implement-auth-code/overview/), [native](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/), [browser](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/) and [service](https://developer.okta.com/docs/guides/implement-client-creds/overview/) type app ...
+| CLI_OKTA_CLIENT_SECRET | The client secret set by Okta for [web](https://developer.okta.com/docs/guides/implement-auth-code/overview/) and [service](https://developer.okta.com/docs/guides/implement-client-creds/overview/) type apps                                              |
+
```
Is this table viewable on mobile?

mraible commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+| CLI_OKTA_CLIENT_ID     | The client id set by Okta for [web](https://developer.okta.com/docs/guides/implement-auth-code/overview/), [native](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/), [browser](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/) and [service](https://developer.okta.com/docs/guides/implement-client-creds/overview/) type app ...
+| CLI_OKTA_CLIENT_SECRET | The client secret set by Okta for [web](https://developer.okta.com/docs/guides/implement-auth-code/overview/) and [service](https://developer.okta.com/docs/guides/implement-client-creds/overview/) type apps                                              |
+
+## Adding Samples for Okta CLI
```
Unfortunately, it's probably best to remove this section. Unless you want to offer an OktaDev t-shirt for creating a PR.

mraible commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+
+Since Hacktoberfest runs off merged pull requests, we'll create the sample in the [okta-samples](https://github.com/okta-samples) GitHub org. You can then put up a pull request to it.
+
+## Wrapping Up
```
```suggestion
## Learn More About Okta, OAuth 2.0, and OpenID Connect
```

mraible reviewed (changes requested):
We might want to add a section with videos since we'll have three when this is published (Spring Boot, Angular, and React Native). Maybe a row with all three thumbnails? I can help with this.

dogeared commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+    "python",
+    "ruby",
+    "security",
+    "devops"
```
It's aspirational - I want to create samples for all of these. Would you suggest I remove some?

dogeared commented on _source/_posts/2020-10-04-introducing-okta-cli.md:
```diff
@@ -0,0 +1,319 @@
+description: "Use the Okt CLI to get started in under 5 minutes!"
+tags: [security, oauth, oauth2, oidc, openid-connect]
+tweets:
+- "Add an Okta integration for your custom app in less than 5 minutes with the Okta CLI"
```
not at this time, but that would be a great flag. Kind of like anyone can host start.spring.io. <- @bdemers

dogeared reviewed (commented):
I addressed all the feedback, rebased master and renamed the file for tomorrow. The tables look reasonable on mobile.

![image](https://user-images.githubusercontent.com/364562/101680936-665a4d00-3a2f-11eb-92ba-8c434e3aaa76.png)

![image](https://user-images.githubusercontent.com/364562/101680961-6e19f180-3a2f-11eb-99e3-fd922e6059fc.png)

dogeared commented:
I addressed all the feedback, rebased master and renamed the file for tomorrow. The tables look reasonable on mobile.

![image](https://user-images.githubusercontent.com/364562/101680936-665a4d00-3a2f-11eb-92ba-8c434e3aaa76.png)

![image](https://user-images.githubusercontent.com/364562/101680961-6e19f180-3a2f-11eb-99e3-fd922e6059fc.png)

mraible commented:
Nice! You might want to copy it into GDoc for editorial or ask @KMDeMat to review this PR and add comments inline.

mraible commented:
@dogeared I added screencasts in [this commit](https://github.com/oktadeveloper/okta-blog/pull/434/commits/22a82b4965d157fbaf517e557da3c1fae4fb0269). @Quorralyne I also added a link to your .NET 5 post. Please let us know if you'd rather have it link to a video (or update this branch with changes).

mraible commented on _source/_posts/2020-12-10-introducing-okta-cli.md:
```diff
@@ -51,7 +51,7 @@ brew cask install oktadeveloper/tap/okta
 Pretty easy, right?
 
-## Get Started in Under Five Minutes
+## Get Started in Under 5 Minutes
```
I think it looks better as a number in the title. :)
